File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(strand1, strand2)
    if strand1.length == strand2.length
      compare(strand1, strand2)
    else
      raise ArgumentError
    end
  end

  def self.compare(strand1, strand2)
    strand1.chars.zip(strand2.chars).select do |n1, n2|
      n1 != n2
    end.count
  end
end

File: javascript/hamming/hamming.js
function Hamming() {

}

Hamming.prototype.compute = function ( input1, input2 ) {
  var count = 0;

  if( input1.length !== input2.length ) {
    throw new Error('DNA strands must be of equal length.');
  }

  for( var i = 0; i < input1.length; i++ ) {
    if( input1.charAt(i) !== input2.charAt(i) ) {
      count++;
    }
  }
  return count;
};

module.exports = Hamming;
